### PR TITLE
Add recommended event

### PR DESCRIPTION
**Pull request by samyasnoun**
Add Recommended Event Interactor and Associated Components

---

### Description
This pull request introduces the functionality for adding recommended events to a group's calendar. The following components have been added:
- **AddRecommendedEventDataAccessInterface**: Interface for retrieving the group from the data layer.
- **AddRecommendedEventInputBoundary**: Interface for the input boundary of the Add Recommended Event use case.
- **AddRecommendedEventOutputBoundary**: Interface for the output boundary, responsible for presenting the success of the event addition.
- **AddRecommendedEventInputData**: Input data containing the event and the group name to be added.
- **AddRecommendedEventOutputData**: Output data containing the event name once added.
- **AddRecommendedEventInteractor**: The interactor that implements the logic for adding the recommended event to the group calendar.

This functionality addresses the need for programmatically adding events suggested to a group, ensuring the group calendar is updated accordingly.

---

### Changes
- [x] Created the **AddRecommendedEventDataAccessInterface** for retrieving a group by its name.
- [x] Implemented the **AddRecommendedEventInputBoundary** to define the execution contract for adding a recommended event.
- [x] Added the **AddRecommendedEventOutputBoundary** to handle success responses.
- [x] Defined the **AddRecommendedEventInputData** class to hold the input for the interactor (event and group name).
- [x] Created the **AddRecommendedEventOutputData** class to encapsulate the event name as the output.
- [x] Developed the **AddRecommendedEventInteractor** to handle the logic of adding the recommended event to the group's calendar.

---

### Checklist
- [x] Code is clean and follows the project's coding standards
- [x] Documentation has been updated (if necessary)
- [x] All tests pass locally (if applicable)

---

### Additional Comments
This PR implements the full functionality for adding recommended events to a group's calendar based on the provided event data. The system ensures that the event is added only if the group exists and is successfully saved to the calendar. The output boundary provides a response containing the name of the event added.

### CODE REVIEW

**Review by CulmoneY — CHANGES_REQUESTED** [review]
This is not the intended logic for the usecase. All this usecase should do is add the event in the input data to the database

**Review by CulmoneY — DISMISSED** [review]
This part looks good. Continue with the presenter, controller and DAO

File: src/daos/MongoDAO.java [hunk 1]
```diff
@@ -576,4 +577,35 @@ public void removeFriend(String userId, String friendId) {
...
+
+
+        // Step 3: Get the existing list of events or initialize a new one
+        List<Document> eventDocs = (List<Document>) calendarDoc.get("events");
+        if (eventDocs == null) {
+            eventDocs = new ArrayList<>();
+        }
+
+        // Step 4: Create a new event document
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```
Comment (Lakssito) [inline src/daos/MongoDAO.java:598]: This Formatter is called a lot in other methods. Please refractor it and make this a instance variable.

File: src/interface_adapter/AddRecommendedEvent/AddRecommendedEventPresenter.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import usecases.add_recommended_event.AddRecommendedEventOutputBoundary;
+import usecases.add_recommended_event.AddRecommendedEventOutputData;
+
+public class AddRecommendedEventPresenter implements AddRecommendedEventOutputBoundary {
+    private final ViewManagerModel viewManagerModel;
+    private final AddRecommendedEventViewModel addRecommendedEventViewModel;
+
+    public AddRecommendedEventPresenter(ViewManagerModel viewManagerModel, AddRecommendedEventViewModel addRecommendedEventViewModel) {
+        this.addRecommendedEventViewModel = addRecommendedEventViewModel;
+        this.viewManagerModel = viewManagerModel;
```
Comment (Lakssito) [inline src/interface_adapter/AddRecommendedEvent/AddRecommendedEventPresenter.java:14]: The View Manager Model is not used in this presenter. Please remove it.

File: src/interface_adapter/AddRecommendedEvent/AddRecommendedEventState.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package interface_adapter.AddRecommendedEvent;
+
+import entity.Event;
+
+public class AddRecommendedEventState {
+    private String eventName;
+    private String groupName;
```
Comment (Lakssito) [inline src/interface_adapter/AddRecommendedEvent/AddRecommendedEventState.java:7]: GroupName is never called and can thus be removed. Please do so.

**Review by Lakssito — CHANGES_REQUESTED** [review]
Please read my comments and make the required changes.

**Review by Lakssito — DISMISSED** [review]
I approve this changes.

**Review by Lakssito — APPROVED** [review]
Everything looks good to me, I approve the changes.

**Pull request merged**